pyre/extra_tests/snippets/code_object_weakref.py

```
@@ -0,0 +1,31 @@
+
+# A weak-keyed mapping over code objects is the shape callers actually use.
+table = weakref.WeakKeyDictionary()
+table[code] = 'protected'
+assert table[code] == 'protected'
+
+# A compiled-on-the-fly code object has no enclosing `co_consts` holding it,
+# so dropping the last reference must clear the weak reference and fire the
+# callback.  A nested function's code would stay alive through its parent.
+dropped = []
+transient = compile('0', '<weakref probe>', 'eval')
+watcher = weakref.ref(transient, lambda ref: dropped.append(True))
```

Ruff (0.16.1)

[warning] Unused lambda argument: `ref`

(ARG005)